From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 16 Feb 2026 11:52:53 +0100
Subject: overview: Only activate if we have a toplevel
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Avoids a critical when swiping in the overview

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1876>
(cherry picked from commit b0819d53682146673f09995a5ec9ee275eed3580)
---
 src/overview.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/overview.c b/src/overview.c
index 61aea0b..fc74ffe 100644
--- a/src/overview.c
+++ b/src/overview.c
@@ -702,6 +702,11 @@ on_page_changed (PhoshOverview *self, guint index, HdyCarousel *carousel)
   list = gtk_container_get_children (GTK_CONTAINER (carousel));
   activity = PHOSH_ACTIVITY (g_list_nth_data (list, index));
   toplevel = get_toplevel_from_activity (activity);
+
+  /* TODO: Mark activation as pending an activate ones the toplevels shows up */
+  if (!toplevel)
+    return;
+
   phosh_toplevel_activate (toplevel, phosh_wayland_get_wl_seat (phosh_wayland_get_default ()));
 
   if (!gtk_widget_has_focus (GTK_WIDGET (activity)))
